odsApi_versioned_docs/version-7.3/how-to-guides/token-access-monitoring.mdx

```diff
@@ -0,0 +1,438 @@
+---
+description: Sample instructions for logging and monitoring access token usage
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# How To: Client Access Token Monitoring
+
+## Overview
+
+This guide describes how to implement monitoring for client access tokens in the Ed-Fi ODS/API. By tracking token usage, administrators can enhance security monitoring, detect unusual access patterns, and maintain audit trails for compliance purposes.
+
+## Use Case
+
+The Ed-Fi ODS/API's `EdFi_Admin` database manages API client access through a hierarchical structure:
+
+- **Vendors** contain multiple **Applications**
+- **Applications** contain multiple **API Clients**
+- **API Clients** have associated **Client Access Tokens**
+
+Each successful client authentication request generates a client access token in the `dbo.clientaccesstokens` table. Monitoring the usage of these tokens is valuable for security and operational insights. However, monitoring was not included out-of-the-box in the Ed-Fi ODS/API. Without additional logging, it's difficult to:
+
+- Track when and how frequently tokens are being used.
+- Identify potentially compromised tokens showing unusual access patterns.
+- Maintain audit logs for security compliance.
+- Analyze usage patterns for capacity planning.
+
+## Security Benefits
+
+Implementing client access token monitoring provides several security advantages:
+
+:::tip Security Benefits
+
+- **Anomaly Detection**: Identify tokens being accessed at unusual times or frequencies.
+- **Audit Trail**: Maintain comprehensive logs for compliance and forensic analysis.
+- **Breach Detection**: Quickly identify if tokens are being used from unexpected sources.
+- **Usage Analytics**: Monitor API usage patterns to detect potential abuse.
+- **Token Lifecycle Management**: Track token usage to determine when tokens should be rotated.
+
+:::
+
+## Implementation
+
+### Step 1: Create Access Log Table
+
+First, create a table to store access log entries:
+
+  <Tabs>
+    <TabItem value="MSSQL" label="SQL Server" default>
+    ```sql
+    CREATE TABLE dbo.ClientAccessTokenLogs
+    (
+      LogId BIGINT IDENTITY(1,1) NOT NULL PRIMARY KEY,
+      TokenId UNIQUEIDENTIFIER NOT NULL,
+      AccessTime DATETIME2 NOT NULL DEFAULT SYSUTCDATETIME(),
+      ApiClientId INT NOT NULL,
+      client_id VARCHAR(50) NOT NULL,
+      ApplicationId INT NOT NULL,
+      ApplicationName NVARCHAR(MAX) NOT NULL,
+      VendorId INT NOT NULL,
+      VendorName NVARCHAR(MAX) NOT NULL
+    );
+
+    CREATE INDEX idx_clientaccesstoken_logs_access_time
```

Comment:
It would be ideal to keep the same table and index naming conventions as the other tables in the Admin database, under the mssql folder - https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/blob/main/Artifacts/MsSql/Structure/Admin/0040-IdColumnIndexes.sql